neo/Consensus/PrepareResponse.cs

```diff
@@ -1,12 +1,21 @@
 ﻿using System.IO;
+using Neo.Network.P2P.Payloads;
+using Neo.IO;
 
 namespace Neo.Consensus
 {
     internal class PrepareResponse : ConsensusMessage
     {
-        public byte[] Signature;
+        /// <summary>
+        /// Speaker PrepareRequest Payload
+        /// </summary>
+        public ConsensusPayload PreparePayload;
```

Comment:
Why not just use `PrepareRequest`?

```c#
public PrepareRequest Request;
```

Reply:
You check everything!! eahuehauea
Could be, @erikzhang, I think that we do not use the Payload itself, you are right.

But I would need to double check the logic for the Regeneration (as well as when we indirectly receive this Payload OnPrepareResponse), because maybe we need some verification regarding view_number, etc...

If we decide to change this I personally would prefer to merge and then we perform this optimization until we are sure about all consequences.
Sorry for the bad design =/

Reply:
You can get the `view_number` from the `ConsensusPayload` with `PrepareResponse` itself.

I need the code to be merged into the master branch to be well designed. If necessary, perhaps this pr will need to be split into multiple smaller prs to submit.

Reply:
I think `public ConsensusPayload PreparePayload;` & `public PrepareRequest Request;` are kind of the same. 
1. `RrepareRequest `is inherited from `ConsensusPayload `. 
2. `PreaparePayload `is used the same way as `PrepareRequest`. 

@vncoelho I think technically  PreparePayload is the same as PrepareRequest, right ?

Reply:
Yes, you are right, `view_number` can be obtained.
Furthermore, the other data there can also be modified, because the signature currently does not cover that variables.

The idea is, should we change it now or wait for a next PR?

Reply:
@erikzhang, I think that we will also need the `payload.ValidatorIndex`, in particular, when regenerating, it will help node to know who was the Speaker.
But we can try to guess as well.

Reply:
https://github.com/igormcoelho/neo/pull/4

Reply:
Please considering that `PreparePayload `has more info than `PrepareRequest`, such as PrevHash, BlockIndex, TimeStamp.  Question is that do we need to verify those info when we consider the requests are IDENTICAL?

Reply:
@erikzhang, I am still having problems with https://github.com/igormcoelho/neo/pull/4

It is a good idea and should work, but need some adjustments.
Do you think we should try to merge it now or it can be done after this one?